chore: Normalize how `resources` are set for each container

lindhe opened the pull request:
### Description

This PR normalizes how we set `resources` for each container and initContainer, first by standardizing how we include the values in the templates and secondly by resouces to `{}` by default, aligned with Bitnami's standard.

Here's one example of Bitnami documenting their reasoning for it:

> We usually recommend not to specify default resources and to leave this as a conscious choice for the user. This also increases chances charts run on environments with little resources, such as Minikube.

— https://github.com/bitnami/charts/blob/7345c0bb2842b73f299aacd2af37147bd7b714cd/bitnami/nginx/values.yaml#L331-L334

We don't have to follow Bitnami's standard, but I think it's reasonable to do so. Unless we are very confident that a container has some upper/lower bound on its resource utilization, it's probably best to condition the chart user to always set the resources explicitly.

Considering how Kubernetes hide the exact capacity of CPU cores etc., it's pretty much impossible to set resource requests/limits that work the same for everyone. This strategy also reduces the need to make additional changes when updating the chart's default images in the future.

Fixes #109 

### Checklist

Ticks in all boxes and 🟢 on all GitHub actions status checks are required to merge:

- [x] I'm aware of the [code of conduct](https://inveniordm.docs.cern.ch/contribute/code-of-conduct/).
- [x] I've created [logical separate commits](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#commits) and followed the [commit message format](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#commit-message).
- [ ] I've added relevant test cases.
- [x] I've added relevant documentation.
- [ ] I've marked [translation strings](https://inveniordm.docs.cern.ch/develop/howtos/i18n/).
- [ ] I've identified the [copyright holder(s)](https://inveniordm.docs.cern.ch/contribute/copyright-policy/) and updated copyright headers for touched files (>15 lines contributions).
- [x] I've NOT included third-party code (copy/pasted source code or new dependencies).
    * If you have added [third-party code (copy/pasted or new dependencies)](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#third-party-codedependencies), please reach out to an [architect on Discord](https://discord.gg/8qatqBC).

**Frontend**

- [ ] I've followed the [CSS/JS](https://inveniordm.docs.cern.ch/develop/best-practices/css-js/) and [React](https://inveniordm.docs.cern.ch/develop/best-practices/react/) guidelines.
- [ ] I've followed the [web accessibility](https://inveniordm.docs.cern.ch/develop/best-practices/accessibility/) guidelines.
- [ ] I've followed the [user interface](https://inveniordm.docs.cern.ch/develop/best-practices/ui/) guidelines.


**Reminder**

By using GitHub, you have already agreed to the [GitHub’s Terms of Service](https://help.github.com/articles/github-terms-of-service/#6-contributions-under-repository-license) including that:

1. You license your contribution under the same terms as the current repository’s license.
2. You agree that you have the right to license your contribution under the current repository’s license.

lindhe commented:
There may be additional `initContainers` that I have not considered. But I think those has to be postponed until later, because it's practically impossible to work with `initContainers` in the chart right now. The chart has to be restructured w.r.t. initContainers before it makes any sense to spend time on them, in my opinion.

egabancho reviewed (approved):
Just to questions:
- Could we comment similarly to the one you linked on the nginx chart? `We usually recommend not to specify default resources and to leave this as a conscious...`
- Since we are, essentially, removing default values for resources, do you think there is a way to warn users about this change? Other than release notes ☺️

lindhe commented:
@egabancho I've updated the comments now, so I think that's been resolved.

I've also added some logic to NOTES for printing a warning if `resources` is unset for any of our internal resources (not dependencies; I'm leaving that for Bitnami to implement). Please take a look and see if you like the solution. It turned out to be more complex that I first thought it would be, so maybe I've missed something obvious.

Resulting message looks like this:

![image](https://github.com/user-attachments/assets/3b222aa9-b543-4810-ad52-3686fc731498)

egabancho commented:
LGTM!
